events/equeue/equeue_mbed.cpp

```diff
@@ -177,23 +182,21 @@ static void equeue_sema_timeout(equeue_sema_t *s)
 
 bool equeue_sema_wait(equeue_sema_t *s, int ms)
 {
-    int signal = 0;
     ALIAS_TIMEOUT timeout;
-    if (ms == 0) {
-        return false;
-    } else if (ms > 0) {
+    if (ms > 0) {
         timeout.attach_us(callback(equeue_sema_timeout, s), (us_timestamp_t)ms * 1000);
     }
 
     core_util_critical_section_enter();
-    while (!*s) {
+    while (!*s && ms != 0) {
         sleep();
         core_util_critical_section_exit();
+        __ISB();
```

Comment:
Relatedly, does this still need to be special? Or can we delegate to the new `do_timed_sleep_absolute` somehow?

Reply:
You're right, that is a halfway house I hadn't thought of - that is available without the RTOS API.

If we did decide to keep two paths, conditional on RTOS API, then yes, would make sense to use it.